File: 9-06-21/src/com/example/Employee.java
package com.example;

public class Employee {
	
	private int employeeId;
	private String firstName;
	private String lastName;
	private String email;
	private double salary;
	public Employee() {
		super();
	}


	public Employee(int employeeId, String firstName, String lastName, String email, double salary) {
		super();
		this.employeeId = employeeId;
		this.firstName = firstName;
		this.lastName = lastName;
		this.email = email;
		this.salary = salary;
	}


	@Override
	public String toString() {
		return "Employee [employeeId=" + employeeId + ", firstName=" + firstName + ", lastName=" + lastName + ", email="
				+ email + ", salary=" + salary + "]";
	}


	@Override
	public boolean equals(Object obj) {
		Employee employee=null;
		if(obj instanceof Employee)
		{
			employee=(Employee)obj;
		}
		if(this.salary==employee.salary)
		{
			return true;
		}
		else
		{
			return false;
		}
		
		
	}
	

}

File: javascript_example/employee.js
class employee
{
    _firstName;
    _lastName;
    _employeeId;
    _email;

    constructor(_firstName,_lastName,_employeeId,_email)
    {
        this._firstName=_firstName;
        this._lastName=_lastName;
        this._employeeId=_employeeId;
        this._email=_email;
        console.log('within parameter constructor');
    }
    getDetails(){
        return 'Employee ID: '+this._employeeId+' First Name: '+this._firstName+' Last Name: '+this._lastName+' Email: '+this._email;
    }
}
class permanentEmployee extends employee{
    _employeeSal;

    constructor(_employeeSal){
        super('vinita','dodda',100,'vinitha@gmail.com');
        this._employeeSal=_employeeSal
        console.log('permanent constructor');
    
    }
    getDetails(){
        return super.getDetails()+ 'employee sal: '+this._employeeSal;
    }
 }
let emp1=new 

let employee1=new employee('vinni','d','vinni@gmail.com');
console.log(employee1.getDetails());
let employee2=new employee('preethi','pre','preethi@gmail.com');
console.log(employee2.getDetails());

let _cd1=new specialEditionCD('3D');
console.log(_cd1.getDetails());